src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,452 @@
+and an asynchronous function. Since you can only use the `await` keyword within 
+an `async` function body, the example converts the parent function from 
+synchronous to asynchronous thus enabling use of the `await` keyword: 
+
+
+[//]: https://gist.github.com/8f4d136e7eb447f4acc7ccb523006e10
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=8f4d136e7eb447f4acc7ccb523006e10"
+  style="margin-top: 25px; margin-bottom: 25px"
+  frameborder="no"
+  height="425"
+  width="100%">
+</iframe>
+
+In this synchronous example the console immediately prints the messages 
+"Fetching user order..." and "Your order is: Large Latte".
+
+[//]: https://gist.github.com/b8d026d2b23102084534159fafc9d7c6
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=b8d026d2b23102084534159fafc9d7c6"
+  style="margin-top: 25px; margin-bottom: 25px"
+  frameborder="no"
+  height="425"
+  width="100%">
+</iframe>
+
+In this asynchronous example, the console does not print "Your order is: Large
+Latte" until after a four second delay has passed. This asynchronous example has
+only four changes from the preceding synchronous example:
+
+1. Update the return types for `createOrderMessage` and `getUserOrder` from 
+`String` to `Future<String>`.
+2. Add the **`async`** keyword before the method bodies for `createOrderMessage`
+and the `main` method.
+3. Use the **`await`** keyword before calling the asynchronous functions 
+`getUserOrder` and `createOrderMessage`.
+4. Update the body of `getUserOrder` to use `Future.delayed` to mock asynchronous
+operations that fetch the user's order.
+
+
+{{ site.alert.info }}
+  Dart can [infer the Future\<T\> type](https://dart.dev/guides/language/sound-dart#type-inference) for you.
+{{ site.alert.end }}
+
+{{ site.alert.secondary }}
+  Quick review: 
+  * An `async` function returns a future.
+  * You can only use the `async` and `await` keywords within an `async` function body.
+  * To declare an `async` function, add the keyword after the function's parameters but before the function's body.
+  * Use the `await` keyword before invoking an async function to "wait" for the future to complete either with a value or an error.
+{{ site.alert.end }}
+
+## Execution flow with async and await
+
+As of Dart 2.0, functions marked as `async` run synchronously until the first
+`await` keyword. This means that within an `async` function's body, all
+synchronous code immediately executes. The following example demonstrates how 
+execution proceeds within an `async` function body:
+
+[//]: https://gist.github.com/d7abfdea1ae5596e96c7c0203d975dba
+<iframe 
+  src="https://dartpad.dartlang.org/experimental/embed-new.html?id=d7abfdea1ae5596e96c7c0203d975dba"
+  style="margin-top: 40px; margin-bottom: 40px"
+  frameborder="no"
+  height="425"
+  width="100%">
+</iframe>
+
+Notice that the timing of the output shifts if the print statement
```

Comment:
I would suggest that they move the line themselves and delete the following DartPad instance.